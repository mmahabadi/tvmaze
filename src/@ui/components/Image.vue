<script setup lang="ts">
import { defineProps } from 'vue'

defineProps<{ src?: string; alt?: string }>()
</script>
<template>
  <img v-if="src" :src="src" :alt="alt" class="w-full h-full object-cover" />
  <img
    v-else
    class="w-full h-full object-cover"
    src="@/assets/image-placeholder.jpg"
    alt="image-placeholder"
  />
</template>
