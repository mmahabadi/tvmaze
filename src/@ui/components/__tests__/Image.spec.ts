import { mount, type VueWrapper } from '@vue/test-utils'

import { beforeEach, describe, expect, it } from 'vitest'
import Image from '../Image.vue'

describe('Image', () => {
  let wrapper: VueWrapper<unknown>
  const props = { src: 'inception.jpg', alt: 'Inception' }

  beforeEach(() => {
    wrapper = mount(Image, {})
  })
  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders default image and alt when no props are passed', () => {
    const defaultSrc = '/src/assets/image-placeholder.jpg'
    const defaultAlt = 'image-placeholder'
    // const wrapper = mount(Image)
    const img = wrapper.find('img')

    expect(img.attributes('src')).toBe(defaultSrc)
    expect(img.attributes('alt')).toBe(defaultAlt)
  })

  it('renders with correct src and alt attributes', async () => {
    //Arrange
    await wrapper.setProps(props)
    const img = wrapper.find('img')

    //Assert
    expect(img.attributes('src')).toBe(props.src)
    expect(img.attributes('alt')).toBe(props.alt)
  })
})
