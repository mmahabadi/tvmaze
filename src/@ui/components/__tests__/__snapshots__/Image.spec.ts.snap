// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Image > renders correctly 1`] = `"<img class="w-full h-full object-cover" src="/src/assets/image-placeholder.jpg" alt="image-placeholder">"`;
